analysis/01_cr_analysis_dataset.do

```diff
@@ -114,10 +121,20 @@ replace covid_tpp_probable=positive_covid_test_ever if covid_tpp_probable==. & c
 sum, d f
 
 
-/*Flag earliest date under18s vaccinated*/
-gen under18vacc_temp=covid_vacc_date if age<18
+/*Flag earliest date under18s double-vaccinated*/
+gen under18vacc_temp=covid_vacc_second_dose_date if age>=12 & age<=17
 bysort household_id: egen under18vacc=min(under18vacc_temp)
 format under18vacc* %td
+
+
+/*Get date for when all children aged 12-17 double-vaccinated (or missing otherwise)*/
+bysort household_id: egen num_child_eligible = total(age>=12 & age<=17) // count eligible children per household
+bysort household_id: egen num_child_vacc = total(age>=12 & age<=17 & covid_vacc_second_dose_date!=.) // count double-vaccinated eligible children per household
+gen all_child_vacc = num_child_eligible !=0 & num_child_eligible == num_child_vacc // flag individuals/households in which all eligible children are double-vaccinated
+bysort household_id: egen all_under18vacc = max(under18vacc_temp) if all_child_vacc == 1 // for these households, get last child 2nd vaccine date
+format all_under18vacc %td
+
```

Comment:
This looks great.  Can I double check - kids under 12 were not offered vaccination until spring 2022?

Reply:
Here's what I've found....

The JCVI recommended vaccination of all 5-11 year olds on 16th February 2022:
https://www.gov.uk/government/publications/jcvi-update-on-advice-for-covid-19-vaccination-of-children-aged-5-to-11/jcvi-statement-on-vaccination-of-children-aged-5-to-11-years-old

Then vaccines corresponding to that announcement started on 4th April 2022:
https://www.england.nhs.uk/2022/04/nhs-rolls-out-covid-vaccine-to-five-million-5-to-11-year-olds/

However, 5-11 year olds in a clinical risk group or with an immunosuppressed household member could have received a vaccine from late December 2021:
https://www.gov.uk/government/publications/jcvi-update-on-advice-for-covid-19-vaccination-of-children-and-young-people/jcvi-statement-on-covid-19-vaccination-of-children-and-young-people-22-december-2021

I wasn't aware of this last one until now. Do we need to include them in the code above (along with 12-17 year olds) or will it make negligible difference if numbers are small?

Reply:
I suppose as we have their data we should include them, or exclude households if a child 5-11 years was vaccinated. The latter might be easier to write up!